src/SdkCommon/ClientRuntime.Azure/ClientRuntime.Azure/AzureClientExtensions.cs

```diff
@@ -349,7 +349,7 @@ public static partial class AzureClientExtensions
             var statusCode = initialResponse.Response.StatusCode;
             var method = initialResponse.Request.Method;
             if (statusCode == HttpStatusCode.OK || statusCode == HttpStatusCode.Accepted ||
-                (statusCode == HttpStatusCode.Created && method == HttpMethod.Put) ||
+                (statusCode == HttpStatusCode.Created && (method == HttpMethod.Put || method == new HttpMethod("PATCH"))) ||
```

Comment:
@ScottHolden this change is already put in for the next release for ClientRuntime and is being tracked by #3592 
Just curious what scenarios that you have for which you need this fix?
Also all such contributions need test coverage, else the PR will be not be accepted.